src/util/check.h

```diff
@@ -46,15 +46,26 @@ class NonFatalCheckError : public std::runtime_error
 #error "Cannot compile without assertions!"
 #endif
 
+/** Helper for Assert() */
+void assertion_fail(const char* file, int line, const char* func, const char* assertion);
+
 /** Helper for Assert() */
 template <typename T>
-T get_pure_r_value(T&& val)
+inline T&& inline_assertion_check(T&& val, const char* file, int line, const char* func, const char* assertion)
 {
+    if (!val) assertion_fail(file, line, func, assertion);
```

Comment:
nit: I think it is better to use a new line unless it is a trivial single statement like `return` or `continue` or `break` ...
```suggestion
    if (!val) {
        assertion_fail(file, line, func, assertion);
    }
```